inc/classes/class-srm-redirect.php

```diff
@@ -165,25 +174,41 @@ public function maybe_redirect() {
 				}
 
 				$sanitized_redirect_to = esc_url_raw( apply_filters( 'srm_redirect_to', $redirect_to ) );
+				$match['redirect_to']   = $sanitized_redirect_to;
+				$match['redirect_from'] = $redirect_from;
+				$match['status_code']   = $status_code;
+			}
+		}
 
-				do_action( 'srm_do_redirect', $requested_path, $sanitized_redirect_to, $status_code );
+		return $match;
+	}
 
-				if ( defined( 'PHPUNIT_SRM_TESTSUITE' ) && PHPUNIT_SRM_TESTSUITE ) {
-					// Don't actually redirect if we are testing
-					return;
-				}
+	/**
+	 * Check current url against redirects
+	 *
+	 * @since 1.8
+	 */
+	public function maybe_redirect() {
+		$redirect_match = $this->get_redirect_match();
 
-				header( 'X-Safe-Redirect-Manager: true' );
+		if ( ! empty( $redirect_match['redirect_to'] ) ) {
+			do_action( 'srm_do_redirect', $redirect_match['requested_path'], $redirect_match['redirect_to'], $redirect_match['status_code'] );
```

Comment:
after this change the action only fires if redirect_to is set, before this change it looks like it fired regardless - any reason to change that @ostowe?